controller/client/client.go

```diff
@@ -66,77 +90,226 @@ func normalizeConfig(conf *Config) error {
 	return nil
 }
 
-type client struct {
-	url        string
-	authToken  string
-	httpClient *http.Client
+// ListRules returns the list of rules
+func (c *Client) ListRules(filter *api.RuleFilter) (*api.RulesSet, error) {
+	var rulesSet api.RulesSet
+
+	path := rulesPath
+	query := c.filterToStringQuery(filter)
+
+	if query != "" {
+		path += query
+	}
+
+	if err := c.doRequest("GET", path, nil, nil, &rulesSet, http.StatusOK); err != nil {
+		logrus.WithError(err).Warn("Failed to retrieve rules from controller")
+		return nil, err
+	}
+
+	return &rulesSet, nil
 }
 
-func (c *client) ListRules(filter *api.RuleFilter) (*api.RulesSet, error) {
-	var rulesSet api.RulesSet
+// CreateRules creates the rules
+func (c *Client) CreateRules(rulesSet *api.RulesSet) (interface{}, error) {
```

Comment:
Why are we returning `interface{}`? Can't we return a specific struct? Something similar to the one that you define dynamically?:
```golang
	result := &struct {
		IDs []string `json:"ids"`
	}{}
```